searx/settings_loader.py

```diff
@@ -73,6 +73,10 @@ def update_settings(default_settings, user_settings):
             else:
                 default_settings[k] = v
 
+    categories_as_tabs = user_settings.get('categories_as_tabs')
+    if categories_as_tabs:
+        default_settings['categories_as_tabs'] = categories_as_tabs
```

Comment:
`categories_as_tabs` is a dictionary.
Custom categories need a way to define an icon. See https://github.com/searxng/searxng/issues/690 (_Stop hardcoding icons in the Simple theme_).

So the `categories_as_tabs` should be merged in the way than the engines:

```yaml
use_default_settings:
  categories_as_tabs:
    keep_only:
      - general
      - images
```

or 

```yaml
use_default_settings:
  categories_as_tabs:
    remove:
      - music
      - video
      - news
```

The code from [lines 76 to 107](https://github.com/searxng/searxng/blob/50ec821e8c5b099947d657722b1879fc8c03bcfa/searx/settings_loader.py#L76-L107) could be a function. So the same code would deal with `engines` and `category_as_tab`.

Reply:
> The code from [lines 76 to 107](https://github.com/searxng/searxng/blob/50ec821e8c5b099947d657722b1879fc8c03bcfa/searx/settings_loader.py#L76-L107) could be a function.
..
> Implemented in https://github.com/searxng/searxng/commit/0811bc70157bb3d7131910f153ea371c773b3911

Thanks a lot .. it works .. but isn't a function .. I think it is not easy to implement a function of a merge that has 'keep_only', 'remove' and 'update dict' .. at least I get a headache :-)

Reply:
The engines are a list and categories_as_tab is a dictionary, so making a function is cumbersome.